frontend/src/features/material-details/components/DeliveryCreationModal.tsx

```diff
@@ -0,0 +1,418 @@
+                                disabled={mode === 'edit' || !temporaryDelivery?.partnerBpnl}
+                                isOptionEqualToValue={(option, value) => option?.bpns === value.bpns}
+                                onChange={(_, value) =>
+                                    setTemporaryDelivery({
+                                        ...temporaryDelivery,
+                                        ...(direction === DirectionType.Inbound
+                                            ? { originBpns: value?.bpns ?? undefined }
+                                            : { destinationBpns: value?.bpns ?? undefined }),
+                                    })
+                                }
+                                value={
+                                    partners
+                                        ?.find((s) => s.bpnl === temporaryDelivery?.partnerBpnl)
+                                        ?.sites.find((s) =>
+                                            direction === DirectionType.Inbound
+                                                ? s.bpns === temporaryDelivery.originBpns
+                                                : s.bpns === temporaryDelivery.destinationBpns
+                                        ) ?? null
+                                }
+                                label={`${direction === DirectionType.Inbound ? 'Origin' : 'Destination'}*`}
+                                placeholder={`Select a ${direction === DirectionType.Inbound ? 'Origin' : 'Destination'} Site`}
+                                error={
+                                    formError &&
+                                    (direction === DirectionType.Inbound
+                                        ? !temporaryDelivery.originBpns
+                                        : !temporaryDelivery.destinationBpns)
+                                }
+                                data-testid="delivery-partner-bpns-field"
+                            />
+                        </Grid>
+                        <Grid item xs={6}>
+                            <FormLabel>Quantity*</FormLabel>
+                            <Input
+                                hiddenLabel
+                                type="number"
+                                value={temporaryDelivery.quantity ?? ''}
+                                error={formError && (temporaryDelivery?.quantity == null || temporaryDelivery.quantity <= 0)}
+                                onChange={(e) =>
+                                    setTemporaryDelivery((curr) => ({
+                                        ...curr,
+                                        quantity: e.target.value ? parseFloat(e.target.value) : undefined,
+                                    }))
+                                }
+                                sx={{ marginTop: '.5rem' }}
+                                data-testid="delivery-quantity-field"
```

Comment:
```suggestion
                                data-testid="delivery-quantity-field"
                                disabled={delivery?.departureType === 'actual-departure' && delivery?.arrivalType === 'actual-arrival'}
                                
```
Currently the quantity is updateable if both arrivals have been set. Obviously missed that during defintion of the rules for #1007 

I would additionally to disable the edit button (gray out + tooltip) the edit pencil in the deliveries list modal.

Note: you can check with int data for todays delivery for semiconductor as customer

Reply:
Thank you for the feedback. Makes sense. Due to the nature of the constraints that only allow `actual-arrival` when `actual-departure` is set, I'll simplify the condition to just check for `arrivalType === 'actual-arrival'`.